diff --git a/dist/pool/index.mjs b/dist/pool/index.mjs
index a188de480fc70dcdfd1a0ebc550c3a91cd39d77a..20933aae8f01e97ec2ef384694c148f46945cbfc 100644
--- a/dist/pool/index.mjs
+++ b/dist/pool/index.mjs
@@ -7626,6 +7626,8 @@ async function getProjectMiniflare(ctx, project) {
     await forEachMiniflare(project.mf, (mf) => mf.setOptions(mfOptions));
   } else {
     log2.debug(`Reusing runtime for ${project.relativePath}...`);
+    // Set same options to reset DB connections
+    await forEachMiniflare(project.mf, (mf) => mf.setOptions(mfOptions));
   }
   return project.mf;
 }
